daemon/update_linux.go

```diff
@@ -28,5 +28,6 @@ func toContainerdResources(resources container.Resources) libcontainerd.Resource
 	}
 	r.MemoryReservation = uint64(resources.MemoryReservation)
 	r.KernelMemoryLimit = uint64(resources.KernelMemory)
+	r.PidsLimit = uint64(resources.PidsLimit)
```

Comment:
So I read proto3 docs and installed `protoc` with Go protocol buffer plugin.
Ran `protoc --go_out=./ api.proto` from `vendor/github.com/docker/containerd/api/grpc/types`. It generated `api.pb.go` with `PidsLimit` in `UpdateResource` and a method `GetPidsLimit()` for the same type. Along with this, it also modified a lot of other parts of the file. Not sure if we need so many changes.

I'll wait for an assistance on how to generate `api.pb.go` that fits here 🙂